include/aws/s3/s3_client.h

```diff
@@ -200,6 +200,27 @@ struct aws_s3_client_config {
     /* Callback and associated user data for when the client has completed its shutdown process. */
     aws_s3_client_shutdown_complete_callback_fn *shutdown_callback;
     void *shutdown_callback_user_data;
+
+    /**
+     * Set to true to manually manage the flow-control window of each meta request.
+     *
+     * If false, no back-pressure is applied and data will download as fast as possible.
+     *
+     * If true, the flow-control window of each meta request will shrink as
+     * response body data is downloaded (headers do not affect the window).
```

Comment:
Can we clarify the `response body data` a bit more? 

It's set for the whole client. The response for meta requests besides GET will be consider as part of the response body? (List part, list object and etc.)

Reply:
Let's chat. I'm still wrapping my head around everything aws-c-s3 does under the hood.

My intention is: from the user's perspective, they are only adjusting window for the bytes they receive via the `body_callback`.

If aws-c-s3 is doing other requests under the hood, where the user doesn't directly see the response, then aws-c-s3 should be automagically managing the window for those responses

Reply:
Ah, I see you're talking about the special [list object API](https://github.com/awslabs/aws-c-s3/blob/b93780769953f661bcae294cd26fef76562aabe8/include/aws/s3/private/s3_list_objects.h). Our code would handle flow-control here (either disable it, or let data flow as fast as possible).